themes/src/main/resources/theme/base/login/resources/js/kcMultivalued.js

```diff
@@ -0,0 +1,101 @@
+const DATA_KC_MULTIVALUED = 'data-kcMultivalued';
+const KC_ADD_ACTION_PREFIX = "kc-add-";
+const KC_REMOVE_ACTION_PREFIX = "kc-remove-";
+
+function createAddAction(element) {
+    const name = element.getAttribute("name");
+    const action = document.createElement("a")
+    action.setAttribute("id", KC_ADD_ACTION_PREFIX + element.getAttribute("id"));
+    action.setAttribute("href", "#");
+    action.innerText = "Add " + window.keycloak.profile.getAttribute(name).getDisplayName();
+    action.addEventListener("click", ev => {
+        const elements = getInputElementsByName().get(name);
+        const length = elements.length;
+
+        if (length === 0) {
+            return;
+        }
+
+        const lastNode = elements[length - 1];
+        const newNode = lastNode.cloneNode(true);
+        newNode.setAttribute("id", name + "-" + elements.length);
+        newNode.value = "";
+        lastNode.after(newNode);
+        ev.preventDefault();
+
+        render();
+    });
+
+    element.after(action);
+}
+
+function createRemoveAction(element, isLastElement) {
+    const action = document.createElement("a")
+    action.setAttribute("id", KC_REMOVE_ACTION_PREFIX + element.getAttribute("id"));
+    action.setAttribute("href", "#");
+    action.innerText = "Remove";
+
+    if (isLastElement) {
+        action.innerHTML = action.innerText + " | ";
```

Comment:
Never assign `innerHtml`, this can allow for script injection. If you want to assign the text use [`textContent`](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent).